drivers/eeprom/eeprom_fm24w256.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2021 Caspar Friedrich <c.s.w.friedrich@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <devicetree.h>
+#include <devicetree/gpio.h>
+#include <drivers/eeprom.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <logging/log.h>
+
+#define DT_DRV_COMPAT cypress_fm24w256
+
+// 256-Kbit ferroelectric random access memory logically organized as 32K × 8 bit
+#define SIZE KB(32)
+
+LOG_MODULE_REGISTER(fm24w256, CONFIG_EEPROM_LOG_LEVEL);
+
+struct config {
+	struct i2c_dt_spec i2c_spec;
+	struct gpio_dt_spec wp_spec;
+	bool read_only;
+};
+
+static int configure_pin(const struct gpio_dt_spec *spec)
+{
+	if (!spec->port) {
```

Comment:
Shouldn't be `spec` first checked for being NULL (also https://github.com/zephyrproject-rtos/zephyr/pull/37184/files#r690975245)